mifosng-provider/src/main/java/org/mifosplatform/infrastructure/core/domain/MifosPlatformTenant.java

```diff
@@ -7,118 +7,26 @@
+    public MifosPlatformTenant(final Long id, final Long oltpId, final Long reportId, final String tenantIdentifier, final String name,
+            final String timezoneId, final MifosPlatformTenantConnection oltpConnection,
+            final MifosPlatformTenantConnection reportsConnection) {
         this.id = id;
+        this.oltpId = oltpId;
+        this.reportId = reportId;
         this.tenantIdentifier = tenantIdentifier;
         this.name = name;
-        this.schemaName = schemaName;
-        this.schemaServer = schemaServer;
-        this.schemaServerPort = schemaServerPort;
-        this.schemaUsername = schemaUsername;
-        this.schemaPassword = schemaPassword;
         this.timezoneId = timezoneId;
-        this.autoUpdateEnabled = autoUpdateEnabled;
-        this.initialSize = initialSize;
-        this.testOnBorrow = testOnBorrow;
-        this.validationInterval = validationInterval;
-        this.removeAbandoned = removeAbandoned;
-        this.removeAbandonedTimeout = removeAbandonedTimeout;
-        this.logAbandoned = logAbandoned;
-        this.abandonWhenPercentageFull = abandonWhenPercentageFull;
-        this.maxActive = maxActive;
-        this.minIdle = minIdle;
-        this.maxIdle = maxIdle;
-        this.suspectTimeout = suspectTimeout;
-        this.timeBetweenEvictionRunsMillis = timeBetweenEvictionRunsMillis;
-        this.minEvictableIdleTimeMillis = minEvictableIdleTimeMillis;
-        this.maxIntervalBetweenRetries = maxIntervalBetweenRetries;
-        this.maxRetriesOnDeadlock = maxRetriesOnDeadlock;
-    }
-
-    public String databaseURL() {
-        final String url = new StringBuilder("jdbc:mysql://").append(this.schemaServer).append(':').append(this.schemaServerPort)
-                .append('/').append(this.schemaName).toString();
-        return url;
-    }
-
-    public int getMaxActive() {
-        return this.maxActive;
-    }
-
-    public int getMinIdle() {
-        return this.minIdle;
-    }
-
-    public int getMaxIdle() {
-        return this.maxIdle;
-    }
-
-    public int getTimeBetweenEvictionRunsMillis() {
-        return this.timeBetweenEvictionRunsMillis;
-    }
-
-    public int getMinEvictableIdleTimeMillis() {
-        return this.minEvictableIdleTimeMillis;
-    }
-
-    public int getInitialSize() {
-        return initialSize;
-    }
-
-    public boolean isTestOnBorrow() {
-        return testOnBorrow;
-    }
-
-    public long getValidationInterval() {
-        return validationInterval;
-    }
-
-    public boolean isRemoveAbandoned() {
-        return removeAbandoned;
-    }
-
-    public int getRemoveAbandonedTimeout() {
-        return removeAbandonedTimeout;
-    }
-
-    public boolean isLogAbandoned() {
-        return logAbandoned;
-    }
-
-    public int getAbandonWhenPercentageFull() {
-        return abandonWhenPercentageFull;
+        this.oltpConnection = oltpConnection;
+        this.reportsConnection = reportsConnection;
```

Comment:
Why not simply set the reportsConnection to oltpConnection if it is not available? this would ease the usage in all referencing classes?